openshift.py

```diff
@@ -0,0 +1,750 @@
+import kubernetes
+import pprint
+import logging
+import requests
+import json
+import re
+from flask import Flask, redirect, url_for, request, Response
+
+import sys
+
+# application = Flask(__name__)
+
+
+class openshift:
+    headers = None
+    verify = False
+    url = None
+
+    def __init__(self, url, token, logger):
+        self.set_token(token)
+        self.set_url(url)
+        self.logger = logger
+
+    def set_token(self, token):
+        self.headers = {
+            "Authorization": "Bearer " + token,
+            "Accept": "application/json",
+            "Content-Type": "application/json",
+        }
+
+    def set_url(self, url):
+        self.url = url
+
+    def get_url(self):
+        return self.url
+
+    def cnvt_project_name(self, project_name):
+        suggested_project_name = re.sub("^[^A-Za-z0-9]+", "", project_name)
+        suggested_project_name = re.sub("[^A-Za-z0-9]+$", "", suggested_project_name)
+        suggested_project_name = re.sub("[^A-Za-z0-9\-]+", "-", suggested_project_name)
+        return suggested_project_name
+
+    def get_request(self, url, debug=False):
+        r = requests.get(url, headers=self.headers, verify=self.verify)
+        if debug == True:
+            self.logger.info("url: " + url)
+            self.logger.info("g: " + str(r.status_code))
+            self.logger.info("g: " + r.text)
+        return r
+
+    def del_request(self, url, payload, debug=False):
```

Comment:
Do `DELETE` requests ever have a payload? I thought that in general they were like `GET` requests (URI only, no payload).

Reply:
Actually yes, some DELETE request have payloads, and the delete will not happen if certain fields are not there.

For example deleting the identity, does require a DeleteOptions object that has a non-zero value for gracePeriodSeconds.